### DISCUSSION TITLE
[A11Y] Using DrawingView with screen readers TalkBack and VoiceOver

### DISCUSSION THREAD

**Discussion by towebo** [CommunityToolkit/Maui#2382] [Q&A]
Hi!
I’m visually impaired and use screen readers to interact with my devices.
It’s easy to create accessible apps in MAUI but now I’d like to implement signature capturing with the DrawingView as shown in multiple samples.
My question is if anyone know how to enable direct touch so the user can draw with a finger even if TalkBack or VoiceOver is active?
I know there’s direct touch inn iOS with VoiceOver but haven’t been able to find how to enable it in MAUI. I have no knowledge if there’s similar feature in TalkBack.

The non elegant solution is that the user tuns of the screen reader during the signing but it would be really nice if I could avoid that.

Regards

Karl-Otto

**Comment by VladislavAntonyuk** [comment]
Hello,
Could you a bit clarify the issue? Does VoiceOver somehow conflict with the Drawing View? Does the same issue exist with MAUI GraphicsView?

**Comment by towebo** [comment]
Ok, let's start from the beginning.
A screen reader acts as an overlay of your touch screen and when you place your finger on it you explore what's on the screen. There's a screen reader cursor that you move with swipe gestures or by sliding your finger on the screen. If you put your finger on a button the screen reader cursor is placed on the button. If you want to interact with the button and tap it you have to make sure the screen reader cursor is on the button and then double tap the screen. The double tap can be performed anywhere on the screen, it's the control that the screen reader cursor is on that receives the action.

With this knowledge you can understand that touches on the screen gives different results with and without a screen reader active.

In iOS there's a mode called Direct Touch that passes touches through just as if the screen reader wasn't active instead of capturing it which is the standard behavior.

My question is if someone knows how to activate direct touch for the whole screen or a part of it within a MAUI app. The specific use case is to have the DrawingView that is a Direct Touch area so the user can write a signature even if they have a screen reader active.

**Comment by stephenquan** [comment]
I think I reproed your problem:

 1. Created a new .NET MAUI app
 2. Add the [DrawingView](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/views/drawingview)
 3. Run the app in Windows
 4. Test that DrawingView works (I have a touch screen PC and it responds to my finger touches)
 5. Turn on Windows Narrator
 6. Try the DrawingView again, but, this time it is no longer responding to my finger touches (N.B. mouse works. The touch gestures do not)
 7. Turn off Windows Narrator
 8. The DrawingView now responds to my finger touches again

**Comment by towebo** [comment]
Thanks for confirming this.
It would be nice if someone had an idea on how to resolve this.

**Comment by stephenquan** [comment]
I found that the Narrator also interferes with the following:
 - Slider
 - PanGestureRecognizer
 - PinchGestureRecognizer
 - PointerGestureRecognizer
 - TapGestureRecognizer
